worker-document/src/main/java/com/hpe/caf/worker/document/DocumentMessageProcessor.java

```diff
@@ -81,6 +81,8 @@ public final WorkerResponse doWork() throws InterruptedException, TaskRejectedEx
                 // Process the task
                 processTask();
 
+                // Unload the customization scripts
+                documentWorkerTask.unloadScripts();
```

Comment:
The way you've placed this if there is an exception processing the task, i.e. thrown from `processTask()`, then you will not unload the scripts.